source/2015-09-08-meteor-and-three-js.html.markdown

```diff
@@ -0,0 +1,230 @@
+---
+title: Meteor and THREE.js
+date: 2015-09-08 21:30
+tags:
+---
+
+We at PANTER already combined Meteor with a 2d-canvas-libraries ([Konva / Kinetic.js](https://github.com/konvajs/konva)), so 3d-rendering is the next logical step.
+
+## Introduction
+
+Meteor is a "fullstack" webframework, covering both client and server. If you haven't heard about it, check it out at [meteor.com](https://www.meteor.com/).
+
+Meteor update its views as soon as the underlying data changes in a reactive way. The default rendering engine *Blaze* and the template-language "spacebars" provide an easy way to work with html (btw. official *react*-support coming soon), but what if we want to use other renderers such as THREE.js and update a 3D-View in a reactive way?
+
+## Preparations
+
+I use the meteor-smartpackage 'limemakers:three' in this example, but you can add the THREE.js-files also manually.
+
+## Set the scene
+
+Let's start by defining a first template and attach data (a cursor) to it:
+
+~~~
+
+<template name="app">
+  <h1>my first 3d-App</h1>
+  {{>THREE_scene objects=players width="100%" height="500px"}}
+</template>
+
+Template.app.helpers
+  players: -> Collections.Players.find()
+
+~~~
+
+The *THREE_scene*-template defines a container which is later used by THREE.js to render the scene:
+
+~~~
+
+<template name="THREE_scene">
+  <div class="container" style="width: {{width}};height:{{height}}">
+  </div>
+</template>
+
+Template.THREE_scene.onRendered ->
+  # init renderer and scene
+  renderer = new THREE.WebGLRenderer antialias: true
+  renderer.setPixelRatio window.devicePixelRatio
+  renderer.setSize @$(".container").width(), @$(".container").height()
+  @$(".container").append renderer.domElement
+  scene = new THREE.Scene()
+
+  (...)
+
+~~~
+
+## Setup the camera
+
+Let's add a camera and make it rotateable with the mouse and let there be light:
```

Comment:
"rotateable" -> "rotatable"

Hmm we need CI with spell-check and style checker :)